src/CodeGen_Posix.cpp

```diff
@@ -92,6 +92,10 @@ CodeGen_Posix::Allocation CodeGen_Posix::create_allocation(const std::string &na
         if (stack_bytes > target.maximum_buffer_size()) {
             const string str_max_size = target.has_large_buffers() ? "2^63 - 1" : "2^31 - 1";
             user_error << "Total size for allocation " << name << " is constant but exceeds " << str_max_size << ".";
+        } else if (memory_type == MemoryType::LockedCache) {
```

Comment:
The implementation we discussed is that CodeGen_Hexagon (or another pass) can rewrite LockedCache allocations into allocations with a custom new_expr/free_function. This implementation shouldn't require any modifications to CodeGen_Posix.

Reply:
Allocations are moved to CodeGen_Hexagon.cpp. If memoryType is LockedCache, CodeGen_Hexagon.cpp handles it , if not it is moved further down to CodeGen_Posix.cpp